tests/unit/amo/components/TestAddAddonToCollection.js

```diff
@@ -0,0 +1,388 @@
+      const root = render();
+      dispatchSpy.reset();
+
+      const userId = 2;
+      dispatchSignInActions({ store, userId });
+      root.setProps(mapStateToProps(store.getState(), {}));
+
+      sinon.assert.calledWith(dispatchSpy, fetchUserCollections({
+        errorHandlerId: root.instance().props.errorHandler.id, userId,
+      }));
+    });
+
+    it('does not fetch user collections when signed out', () => {
+      dispatchClientMetadata({ store });
+      const dispatchSpy = sinon.spy(store, 'dispatch');
+      render();
+
+      sinon.assert.notCalled(dispatchSpy);
+    });
+
+    it('does not fetch collections on first render if they exist', () => {
+      signInAndDispatchCollections();
+      const dispatchSpy = sinon.spy(store, 'dispatch');
+      render();
+
+      sinon.assert.notCalled(dispatchSpy);
+    });
+
+    it('does not fetch collections on update if they exist', () => {
+      signInAndDispatchCollections();
+
+      const dispatchSpy = sinon.spy(store, 'dispatch');
+      const root = render();
+      dispatchSpy.reset();
+
+      // Pretend this is updating some unrelated props.
+      root.setProps({});
+
+      sinon.assert.notCalled(dispatchSpy);
+    });
+
+    it('does not fetch user collections while loading', () => {
+      const addon = createInternalAddon(fakeAddon);
+      const userId = 5543;
+      dispatchSignInActions({ store, userId });
+      store.dispatch(fetchUserCollections({
+        errorHandlerId: 'some-id', userId,
+      }));
+
+      const dispatchSpy = sinon.spy(store, 'dispatch');
+      render({ addon });
+
+      sinon.assert.notCalled(dispatchSpy);
+    });
+  });
+
+  describe('selecting a user collection', () => {
+    const findOption = ({ root, text }) => {
+      const option = root.find('.AddAddonToCollection-option')
+        .filterWhere((opt) => opt.text() === text);
+      expect(option).toHaveLength(1);
+      return option;
+    };
+
+    it('renders a disabled select when loading user collections', () => {
+      const userId = 5543;
+      dispatchSignInActions({ store, userId });
+      store.dispatch(fetchUserCollections({
+        errorHandlerId: 'some-id', userId,
+      }));
+
+      const root = render();
+
+      const select = root.find('.AddAddonToCollection-select');
+      expect(select.props().disabled).toEqual(true);
+      expect(select.html()).toContain('Loading...');
+    });
+
+    it('renders a disabled select when adding add-on to collection', () => {
+      const addon = createInternalAddon(fakeAddon);
+      const userId = 5543;
+      dispatchSignInActions({ store, userId });
+      store.dispatch(addAddonToCollection({
+        addonId: addon.id,
+        userId,
+        collectionId: 321,
+        collectionSlug: 'some-collection',
+        errorHandlerId: 'error-handler',
+      }));
+
+      const root = render({ addon });
+
+      const select = root.find('.AddAddonToCollection-select');
+      expect(select.props().disabled).toEqual(true);
```

Comment:
I think you can do `expect(select).toHaveProp('disabled', true);` here and in a few other places, which just outputs better error messages IIRC.